core/src/main/java/org/mskcc/cbio/oncokb/util/VariantConsequenceUtils.java

```diff
@@ -94,6 +95,46 @@ private static void cacheVariantConsequencesMap() {
         }
     }
 
+    public static VariantConsequence consequenceResolver(String consequenceTerms) {
+        return consequenceResolver(consequenceTerms, null);
+    }
+
+    // The goal is to reserve the details of the consequence if possible, but when consequence conflicts with variant class, we need to decide which one to use
+    public static VariantConsequence consequenceResolver(String consequenceTerms, String variantClass) {
+        if (StringUtils.isEmpty(consequenceTerms) && StringUtils.isEmpty(variantClass)) {
+            return null;
+        }
+        if (consequenceTerms == null) {
+            consequenceTerms = "";
+        }
+        if (variantClass == null) {
+            variantClass = "";
+        }
+
+        VariantConsequence consequence = findVariantConsequenceByTerm(consequenceTerms.contains(",") ? pickConsequenceTerm(consequenceTerms) : consequenceTerms);
+        VariantConsequence variantClassConsequence = variantClass == null ? null : findVariantConsequenceByTerm(variantClass);
```

Comment:
`variantClass` will never be null since on lines 110-112, it is assigned to empty string.